app/code/Magento/Eav/Test/Unit/Model/TypeLocator/SimpleTypeTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Eav\Test\Unit\Model\TypeLocator;
+
+use Magento\Eav\Model\TypeLocator\SimpleType;
+use Magento\Eav\Model\AttributeRepository;
+use Magento\Framework\Webapi\CustomAttribute\ServiceTypeListInterface;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Catalog\Model\ResourceModel\Eav\Attribute;
+use Magento\Framework\Reflection\TypeProcessor;
+
+/**
+ * Class SimpleTypeTest
+ *
+ * @package Magento\Eav\Test\Unit\Model\TypeLocator
+ */
+class SimpleTypeTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @var ObjectManager
+     */
+    private $objectManger;
+
+    protected function setUp()
+    {
+        $this->objectManger = new ObjectManager($this);
+    }
+
+    /**
+     * Test testGetType method
+     *
+     * @param ServiceTypeListInterface|\PHPUnit_Framework_MockObject_MockObject $serviceTypeListMock
+     * @param string $getFrontendInputReturn
+     * @param string $getBackendTypeReturn
+     * @param string $expected
+     * @dataProvider getTypeDataProvider
+     */
+    public function testGetType($serviceTypeListMock, $getFrontendInputReturn, $getBackendTypeReturn, $expected)
+    {
+        /**
+         * @var Attribute|\PHPUnit_Framework_MockObject_MockObject
+         */
+        $attributeMock = $this->getMockBuilder(Attribute::class)
+                                    ->disableOriginalConstructor()
+                                    ->setMethods(['getFrontendInput', 'getBackendType'])
+                                    ->getMock();
+        $attributeMock->expects($this->any())
+                            ->method('getFrontendInput')
+                            ->willReturn($getFrontendInputReturn);
+        $attributeMock->expects($this->any())
+                            ->method('getBackendType')
+                            ->willReturn($getBackendTypeReturn);
+
+        /**
+         * @var AttributeRepository|\PHPUnit_Framework_MockObject_MockObject
+         */
+        $attributeRepositoryMock = $this->getMockBuilder(AttributeRepository::class)
+                                              ->disableOriginalConstructor()
+                                              ->setMethods(['get'])
+                                              ->getMock();
+        $attributeRepositoryMock->expects($this->any())
+                                      ->method('get')
+                                      ->willReturn($attributeMock);
+
+        /**
+         * @var SimpleType|\PHPUnit_Framework_MockObject_MockObject
+         */
+        $simpleType = ($this->objectManger)->getObject(
```

Comment:
Get rid of ObjectManager and `setUp` method.

Reply:
Hi @orlangur. I use ObjectManager to create testable class instance. What should I use instead ?

Reply:
I meant just get rid of property and `setUp` method but here it also makes sense to replace it with `new` operator.